bin/tx-prover/src/prover.rs

```diff
@@ -1,49 +1,30 @@
 use alloc::{boxed::Box, string::ToString};
-use core::cell::RefCell;
 
 use miden_objects::transaction::{ProvenTransaction, TransactionWitness};
 use miden_tx::{TransactionProver, TransactionProverError};
 
-use crate::{generated::api_client::ApiClient, RemoteTransactionProverError};
+use crate::generated::api_client::ApiClient;
 
 // REMOTE TRANSACTION PROVER
 // ================================================================================================
 
 /// A [RemoteTransactionProver] is a transaction prover that sends witness data to a remote
 /// gRPC server and receives a proven transaction.
+///
+/// When compiled for the `wasm32-unknown-unknown` target, it uses the `tonic_web_wasm_client`
+/// transport. Otherwise, it uses the built-in `tonic::transport` for native platforms.
+/// The transport layer connection is established lazily when the first transaction is proven.
+
 #[derive(Clone)]
 pub struct RemoteTransactionProver {
```

Comment:
nit: I'd remove the blank line on line 17.

Also, I would add a sentence or two describing the lazy behavior (i.e., that the connection is not established on initialization).

Reply:
Sorry, I missed this comment. It's now fixed